contentcuration/contentcuration/models.py

```diff
@@ -2590,3 +2591,66 @@ class Meta:
                 name='task_result_signature',
             ),
         ]
+
+
+class BaseFeedback(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    context = models.JSONField()
+
+    # for RecommendationInteractionEvent class contentnode_id represents:
+    # the date/time this interaction happened
+    #
+    # for RecommendationEvent class contentnode_id represents:
+    # time_shown: timestamp of when the recommendations are first shown
+    created_at = models.DateTimeField(auto_now_add=True)
+
+    # for RecommendationsEvent class conntentnode_id represents:
+    # target_topic_id that the ID of the topic the user
+    # initiated the import from (where the imported content will go)
+    #
+    # for ReccomendationsInteractionEvent class contentnode_id represents:
+    # contentNode_id of one of the item being interacted with
+    # (this must correspond to one of the items in the “content” array on the RecommendationEvent)
+    #
+    # for RecommendationsFlaggedEvent class contentnode_id represents:
+    # contentnode_id of the content that is being flagged.
+    contentnode_id = models.UUIDField()
+
+    # These are corresponding values of content_id to given contentNode_id for a ContentNode.
+    content_id = models.UUIDField()
+
+    class Meta:
+        abstract = True
+
+
+class BaseFeedbackEvent(models.Model):
+    user = models.ForeignKey(settings.AUTH_USER_MODEL, on_delete=models.CASCADE)
+    # The ID of the channel being worked on (where the content is being imported into)
+    # or the Channel Id where the flagged content exists.
+    target_channel_id = models.UUIDField()
+
+    class Meta:
+        abstract = True
+
+
+class BaseFeedbackInteractionEvent(models.Model):
+    feedback_type = models.CharField(max_length=50, choices=feedback.FEEDBACK_TYPE_CHOICES)
+    feedback_reason = models.TextField()
```

Comment:
Might be a good idea to set a `max_length` here. It can be arbitrary and large enough to fit a reasonable 'reason'.